aten/src/ATen/core/op_registration/op_registration.h

```diff
@@ -328,60 +300,60 @@ class CAFFE2_API RegisterOperators final {
       } else {
         return std::move(*this).kernel(
           c10::nullopt,
-          KernelFunction::makeFromUnboxedOnlyFunction<FuncType, kernel_func>(),
+          KernelFunction::makeFromUnboxedOnlyRuntimeFunction(kernel_func),
           // TODO Do schema inference without relying on WrapKernelFunction
-          detail::FunctionSchemaInferer<typename detail::WrapKernelFunction<FuncType, kernel_func>::type>()()
+          detail::FunctionSchemaInferer<detail::WrapRuntimeKernelFunctor<guts::decay_t<FuncType>>>()()
         );
       }
     }
 
     // TODO Remove impl_unboxedOnlyC10Kernel once all of aten can generate boxed kernels
-    template<class FuncType, FuncType* kernel_func>
+    template<class FuncType>
     // enable_if: only enable it if FuncType is actually a function
-    guts::enable_if_t<guts::is_function_type<FuncType>::value, Options&&> impl_unboxedOnlyC10Kernel(TensorTypeId dispatch_key) && {
```

Comment:
This one might give you binary size improvements and it can be changed without breaking anything.